# German Ultimate Frisbee Clubs Count per State

**Aim**
- Display number of clubs in each german state on a map

**Method**
- Get data from https://dfv-mv.de/verbaende/1/ using _requests_
- Read out relevant data using _BeautifulSoup_ and _re_
- Display data in map using _Tableau_

In [59]:
import pandas as pd
from bs4 import BeautifulSoup
from requests import get
import re

r = get('https://dfv-mv.de/verbaende/1/')

soup = BeautifulSoup(r.text, 'html.parser')
# print(soup.prettify())
soup.table

<table class="table table-striped table-bordered table-hover">
<thead>
<tr>
<th class="name orderable"><a href="?sort=name">Verband</a></th>
<th class="fon orderable"><a href="?sort=fon">Telefon</a></th>
<th class="orderable verein__count"><a href="?sort=verein__count">Angeschlossene Vereine</a></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td class="name"><a href="/verband/brandenburg/">Brandenburgischer Frisbeesport-Verband e.V.</a></td>
<td class="fon">0172-8252515</td>
<td class="verein__count">3</td>
</tr>
<tr class="even">
<td class="name"><a href="/verband/deutscher-frisbeesport-verband/">Deutscher Frisbeesport-Verband</a></td>
<td class="fon">0221-99374019</td>
<td class="verein__count">24</td>
</tr>
<tr class="odd">
<td class="name"><a href="/verband/hessen/">Frisbeesport Landesverband Hessen e.V.</a></td>
<td class="fon">06142 850830</td>
<td class="verein__count">14</td>
</tr>
<tr class="even">
<td class="name"><a href="/verband/bayern/">Frisbeesport-Landesverband Bayern e.V

*Example of website content:*
<div class="table-responsive">
        <table class="table table-striped table-bordered table-hover">
            <thead>
                <tr>
                            <th class="name orderable"><a href="?sort=name">Verband</a></th>
                            <th class="fon orderable"><a href="?sort=fon">Telefon</a></th>
                            <th class="orderable verein__count"><a href="?sort=verein__count">Angeschlossene Vereine</a></th>
                </tr>
            </thead>
            <tbody>
                    <tr class="odd">
                            <td class="name"><a href="/verband/brandenburg/">Brandenburgischer Frisbeesport-Verband e.V.</a></td>
                            <td class="fon">0172-8252515</td>
                            <td class="verein__count">3</td>
                    </tr>
                    <tr class="even">
                            <td class="name"><a href="/verband/deutscher-frisbeesport-verband/">Deutscher Frisbeesport-Verband</a></td>
                            <td class="fon">0221-99374019</td>
                            <td class="verein__count">24</td>
                    </tr>
                    <tr class="odd">
                            <td class="name"><a href="/verband/hessen/">Frisbeesport Landesverband Hessen e.V.</a></td>
                            <td class="fon">06142 850830</td>
                            <td class="verein__count">14</td>
                    </tr>
                    <tr class="even">
                            <td class="name"><a href="/verband/bayern/">Frisbeesport-Landesverband Bayern e.V.</a></td>
                            <td class="fon">+49 8152 78820</td>
                            <td class="verein__count">33</td>
                    </tr>
                    <tr class="odd">
                            <td class="name"><a href="/verband/berlin/">Frisbeesport-Landesverband Berlin e.V.</a></td>
                            <td class="fon">—</td>
                            <td class="verein__count">10</td>
                    </tr>
                    <tr class="even">
                            <td class="name"><a href="/verband/nordrhein-westfalen/">Frisbeesport-Landesverband Nordrhein-Westfalen e.V.</a></td>
                            <td class="fon">0251-132962</td>
                            <td class="verein__count">41</td>
                    </tr>
                    <tr class="odd">
                            <td class="name"><a href="/verband/baden-wurttemberg/">Frisbeesportverband Baden-Württemberg</a></td>
                            <td class="fon">—</td>
                            <td class="verein__count">27</td>
                    </tr>
                    <tr class="even">
                            <td class="name"><a href="/verband/rheinland-pfalzsaarland/">Frisbeesportverband Rheinland-Pfalz/Saarland e.V.</a></td>
                            <td class="fon">0176-75139025</td>
                            <td class="verein__count">9</td>
                    </tr>
                    <tr class="odd">
                            <td class="name"><a href="/verband/niedersachsen/">Niedersächsischer Frisbeesportverband</a></td>
                            <td class="fon">Tel.: 05371-7430752</td>
                            <td class="verein__count">18</td>
                    </tr>
            </tbody>
        </table>
    </div> <!-- end .table-responsive -->

In [103]:
names = soup.find_all('td', class_='name')
orgnames = []
for name in names:
    orgname = name.find('a')
    orgnames.append(orgname.text)
orgnames

['Brandenburgischer Frisbeesport-Verband e.V.',
 'Deutscher Frisbeesport-Verband',
 'Frisbeesport Landesverband Hessen e.V.',
 'Frisbeesport-Landesverband Bayern e.V.',
 'Frisbeesport-Landesverband Berlin e.V.',
 'Frisbeesport-Landesverband Nordrhein-Westfalen e.V.',
 'Frisbeesportverband Baden-Württemberg',
 'Frisbeesportverband Rheinland-Pfalz/Saarland e.V.',
 'Niedersächsischer Frisbeesportverband']

In [106]:
counts = soup.find_all('td', class_='verein__count')
orgcounts = []
for count in counts:
    orgcounts.append(count.text)
orgcounts

['3', '24', '14', '33', '10', '41', '27', '9', '18']

## Display of Tableau website

#### via IPython
Tableau website not working

#### via iframe
Only allowed in some browsers / with secure browser settings

In [7]:
from IPython.display import HTML
HTML(url='https://public.tableau.com/profile/johannes.sch.ck#!/vizhome/DFV_Vereine_nach_Bundesland/Dashboard1')

In [8]:
%%html
<iframe src='https://public.tableau.com/profile/johannes.sch.ck#!/vizhome/DFV_Vereine_nach_Bundesland/Dashboard1' width=950 height=800></iframe>